tests/chains/Polkadot/TWAnySignerTests.cpp

```diff
@@ -402,38 +490,71 @@ TEST(PolkadotSigner, PolymeshEncodeBondAndNominate) {
-      callIndices->set_module_index(0x29);
-      callIndices->set_method_index(0x02);
+        auto* callIndices = bondnom->mutable_call_indices()->mutable_custom();
+        callIndices->set_module_index(0x29);
+        callIndices->set_method_index(0x02);
     }
 
     // Staking.bond
     {
-      auto* callIndices = bondnom->mutable_bond_call_indices()->mutable_custom();
-      callIndices->set_module_index(0x11);
-      callIndices->set_method_index(0x00);
+        auto* callIndices = bondnom->mutable_bond_call_indices()->mutable_custom();
+        callIndices->set_module_index(0x11);
+        callIndices->set_method_index(0x00);
     }
 
     // Staking.nominate
     {
-      auto* callIndices = bondnom->mutable_nominate_call_indices()->mutable_custom();
-      callIndices->set_module_index(0x11);
-      callIndices->set_method_index(0x05);
+        auto* callIndices = bondnom->mutable_nominate_call_indices()->mutable_custom();
+        callIndices->set_module_index(0x11);
+        callIndices->set_method_index(0x05);
     }
 
-    auto preImage = Signer::signaturePreImage(input);
+    /// Step 2: Obtain preimage hash
+    auto txInputData = data(input.SerializeAsString());
+    auto txInputDataPtr = WRAPD(TWDataCreateWithBytes(txInputData.data(), txInputData.size()));
+    const auto preImageHashes = WRAPD(TWTransactionCompilerPreImageHashes(coin, txInputDataPtr.get()));
+    auto preImageHash = data(TWDataBytes(preImageHashes.get()), TWDataSize(preImageHashes.get()));
+
+    TxCompiler::Proto::PreSigningOutput preSigningOutput;
+    ASSERT_TRUE(preSigningOutput.ParseFromArray(preImageHash.data(), int(preImageHash.size())));
+    ASSERT_EQ(preSigningOutput.error(), Common::Proto::OK);
+
+    const auto preImage = data(preSigningOutput.data());
+
     ASSERT_EQ(hex(preImage), "2902081100005ccc5c9276ab7976e7c93c70c190fbf1761578c07b892d0d1fe65972f6a290610224f4000011050400c6766ff780e1f506e41622f7798ec9323ab3b8bea43767d8c107e1e920581958150300006a995b00040000006fbd74e5e1d0a61d52ccfe9d4adaed16dd3a7caa37c6bc4d0c2fa12e8b2f4063ab67744c78f1facfec9e517810a47ae23bc438315a01dac5ffee46beed5ad3d8");
 
+    // The public key is an SR25519 key and the signature is an SR25519 signature.
     auto publicKey = parse_hex("5ccc5c9276ab7976e7c93c70c190fbf1761578c07b892d0d1fe65972f6a29061");
     auto signature = parse_hex("685a2fd4b1bdf7775c55eb97302a0f86b0c10848fd9db3a7f6bbe912c4c2fa28bed16f6032852ec14f27f0553523dd2fc181a6dca79f19f9c7ed6cb660cf6480");
-    auto encoded = Signer::encodeTransaction(input, publicKey, signature);
-    ASSERT_EQ(hex(encoded), "d90284005ccc5c9276ab7976e7c93c70c190fbf1761578c07b892d0d1fe65972f6a2906100685a2fd4b1bdf7775c55eb97302a0f86b0c10848fd9db3a7f6bbe912c4c2fa28bed16f6032852ec14f27f0553523dd2fc181a6dca79f19f9c7ed6cb660cf6480150300002902081100005ccc5c9276ab7976e7c93c70c190fbf1761578c07b892d0d1fe65972f6a290610224f4000011050400c6766ff780e1f506e41622f7798ec9323ab3b8bea43767d8c107e1e920581958");
```

Comment:
Note that the expected value (signed transaction) here didn't match the original from Polymesh Mainnet here: https://polymesh.subscan.io/extrinsic/0xd516d4cb1f5ade29e557586e370e98c141c90d87a0b7547d98c6580eb2afaeeb
Because that signed transaction used an SR25519 signature and there is no support for SR25519 here.

I have copied the original signed transaction from Mainnet and disabled compiling this transaction, since only ED25519 signatures are supported.  The only difference between these two is one byte (the variant index of `MultiSignature`, ED25519=0, SR25519=1).

Reply:
Thank you for investigating! Makes sense to me